### PR TITLE
feat: support starters

### LINKED ISSUE #247: When devfile component is created from an empty folder, handle starters
[kind/Enhancement]

## Which functionality do you think we should update/improve?

When creating a component from an empty folder, we should look if the devfile has starters and if:
- if there is a single one ask to use or not starter
- if there are several starter, ask to user starter and user should select which one.

## Why is this needed? 

For better bootstrapping

### PR DESCRIPTION
**Pull request by sbouchet**
## What is the purpose of this change? What does it change?
propose starters when the selected context is empty

## Was the change discussed in an issue?
fixes #247 

## How to test changes?
Create an empty folder and select it when creating a new component.

### CODE REVIEW

File: src/main/java/org/jboss/tools/intellij/openshift/utils/odo/OdoCli.java [hunk 1]
```diff
@@ -432,6 +426,8 @@ public void logout() throws IOException {
     public List<ServiceInstance> getServices(String project, String application) {
         try {
             ServiceCatalogClient sc = client.adapt(ServiceCatalogClient.class);
+            client.services().inNamespace(project).withLabelSelector(new LabelSelectorBuilder().addToMatchLabels(KubernetesLabels.APP_LABEL, application).build()).list().getItems();
```
Comment (jeffmaury) [inline src/main/java/org/jboss/tools/intellij/openshift/utils/odo/OdoCli.java:429]: Not sure to understand what is the purpose of this call ?

**Review by jeffmaury — CHANGES_REQUESTED** [review]
Seems to me that:
- as soon as a component type has a starter then it is selected whatever the content of the selected module is
- then there is no way to unselect it
openshift

File: src/main/java/org/jboss/tools/intellij/openshift/utils/odo/OdoCli.java (hunk 1 above)
Comment (sbouchet) [reply src/main/java/org/jboss/tools/intellij/openshift/utils/odo/OdoCli.java:429]: good catch. i'll remove this.

**Comment by sbouchet** [comment]
@jeffmaury : i've fixed the issue when switching context with not empty folder. can you review again ?

**Review by jeffmaury — CHANGES_REQUESTED** [review]
Better but when an empty project is selected, the first starter is automatically selected and there is no way to unselect it. In JBoss Tools we display a message to the user that he can use starters but by default none is selected without user action

**Review by jeffmaury — APPROVED** [review]
LGTM. The only issue (but we have it also in JBT) is that once user has selected a starter, there is no way to reset to no starter

**Comment by sbouchet** [comment]
> LGTM. The only issue (but we have it also in JBT) is that once user has selected a starter, there is no way to reset to no starter

filled #293 to implement it.

**Pull request merged**